pkg/pool-utils/contracts/RelayedBasePool.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/BalancerErrors.sol";
+
+import "./BasePool.sol";
+import "./interfaces/IBasePoolRelayer.sol";
+
+/**
+ * @dev Base Pool associated with a relayer that guarantees it can only be joined/exited from the relayer itself.
+ * This contract as a simple mixin for pools. Implementing pools must make sure to call the BasePool's constructor
+ * properly, and validate that the given
+ */
+abstract contract RelayedBasePool is BasePool {
+    using Address for address;
+
+    IBasePoolRelayer public immutable relayer;
+
+    modifier ensureRelayer(bytes32 poolId) {
+        // Note that this modifier is relying on an input parameter, which is usually given by the Vault which is
+        // considered a trusted party. It's up to the implementing pool whether to validate this value or not.
```

Comment:
I get the intent behind this comment, but it is not entirely accurate. We're inheriting from `BasePool` here and calling its implementation, which _does_ validate `poolId`. I'd consider removing it.